lesson09.md

```diff
@@ -0,0 +1,387 @@
+me.name = "Hasan"
+console.log(me.name)
+```
+
+* What's the result?
+
+```js
+Ownen
+Berlin
+Hasan
+```
+<!-- .element: class="fragment" -->
+
+---
+
+### Recap: Exercise
+
+Using your "me" object, create a method to introduce yourself (using console.log). Create a object within that represents your address.
+
+```js
+me.introduce();
+// "Hi, my name is Owen"
+console.log(me.address.city); // "Berlin"
+console.log(me.address.country); // "Germany"
+```
+
+---
+
+### Solution
+
+```js
+let me = {
+    name: "Owen",
+    introduce: function() {
+        console.log("Hi, my name is " + this.name);
+    },
+    address: {
+        city: "Berlin",
+        country: "Germany"
+    }
+};
+
+me.introduce();
+console.log(me.address.city);
+console.log(me.address.country);
+```
+
+---
+
+### "this" in JavaScript
+
+In JavaScript, ```this```  is a special keyword that refers to the object that the current function is a method of.
+
+---
+
+### Example
+
+```js
+let person = {
+    name: "Hasan",
+    greet: function() {
+        console.log("Hello, my name is " + this.name);
+    }
+};
+
+person.greet();
+```
+
+* What's the result?
+
+```js
+Hello, my name is Hasan
+```
+<!-- .element: class="fragment" -->
+
+---
+
+### Exercise
+
+Create an object car with properties ```company``` and ```model```. Add a method ```describe``` that prints out something like the following:
+
+```js
+"I drive a {company} {model}"
+```
+
+---
+
+### HTML and Javascript
+
+HTML (Structure): HTML provides the basic structure of a webpage. It defines elements like headings, paragraphs, images, and more. It's like the skeleton of a webpage.
+
+```html
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Simple Page</title>
+</head>
+<body>
+    <h1 id="greeting">Hello, World!</h1>
+<body>
+</html>
+```
+---
+
+### HTML and Javascript
+
+Javascript (Interactions): Javascript enables interactivity on a webpage. It allows you to respond to events like clicks, input, and more. JavaScript can change the content, style, and behavior of the webpage dynamically.
+
+```js
+function changeText() {
+  document.getElementById("greeting").textContent = "Hi!";
+}
+```
+
+---
+
+### Example
+
+```html
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Simple Page</title>
+</head>
+<body>
+    <h1 id="greeting">Hello, World!</h1>
+    <button onclick="changeText()">Click Me</button>
+
+    <script>
+        function changeText() {
+            document.getElementById("greeting").textContent = "Hi, there!";
+        }
+    </script>
+</body>
+</html>
+
+```
+
+---
+
+### Onclick Event
+
+The ```onclick``` event is a specific type that happens when a user clicks on an element, like a button, link, or image.
+
+---
+
+### Example
+
+```html
+<button onclick="sayHello()">Click Me</button>
+```
+
+In this example, When a user clicks the button, it triggers the sayHello function
+
+```js
+function sayHello() {
+    alert('Hello, World!');
+}
+```
+
+
+---
+
+### Changing the background color
```

Comment:
We already did that exercise, I think you can safely remove this and the next two slides, which should give us more time to focus on `getElementById`